src/components/DopplerPlus/PlanCalculator.js

```diff
@@ -1,42 +1,67 @@
-import React, { useReducer } from 'react';
+import React, { useReducer, useEffect, useState } from 'react';
 import { Slider } from '../shared/Slider/Slider';
+import { InjectAppServices } from '../../services/pure-di';
+import { Loading } from '../Loading/Loading';
+import { useRouteMatch } from 'react-router-dom';
+import { useIntl } from 'react-intl';
 
-const PlanCalculator = () => {
-  // TODO: get data from a double service
-  const plansList = [
-    { idPlan: 1, price: 15, amount: 1500 },
-    { idPlan: 2, price: 29, amount: 2500 },
-    { idPlan: 3, price: 48, amount: 5000 },
-    { idPlan: 4, price: 77, amount: 10000 },
-    { idPlan: 5, price: 106, amount: 15000 },
-    { idPlan: 6, price: 145, amount: 25000 },
-    { idPlan: 7, price: 240, amount: 50000 },
-    { idPlan: 9, price: 340, amount: 75000 },
-    { idPlan: 9, price: 460, amount: 100000 },
-  ];
+const PlanCalculator = ({ dependencies: { dopplerLegacyClient } }) => {
+  const { params } = useRouteMatch();
+  const { typePlanId, promoId, discountId } = params;
+  const safePromoId = promoId || '';
+  const safeDiscountId = discountId || 0;
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
 
-  const discountsList = [
-    { id: 1, percent: 0, monthsAmmount: 1, description: 'Mensual' },
-    { id: 2, percent: 5, monthsAmmount: 3, description: 'Trimestral' },
-    { id: 3, percent: 15, monthsAmmount: 6, description: 'Semestral' },
-    { id: 4, percent: 25, monthsAmmount: 12, description: 'Anual' },
-  ];
+  const [state, setState] = useState({ loading: true });
 
-  const plansTooltipDescriptions = plansList.map((plan) => {
-    return plan.amount + ' Suscriptores';
-  });
+  const [currentPlan, setSelectedPlan] = useReducer(
+    (currentPlan, index) => {
+      return state.planList[index] || currentPlan;
+    },
+    { idPrice: 0, price: 0, amount: 0 },
+  );
+
+  const [currentDiscount, applyDiscount] = useReducer(
+    (discount, index) => {
+      return state.discountsList[index] || discount;
+    },
+    { id: 0, percent: 0, monthsAmmount: 0, description: '' },
+  );
 
-  const initialPlan = plansList[0];
+  useEffect(() => {
+    const fetchData = async () => {
+      setState({ loading: true });
+      const responsePlansList = await dopplerLegacyClient.getPlansList(typePlanId);
+      if (responsePlansList.success) {
+        setState({
+          loading: false,
+          planList: responsePlansList.planList,
+          discountsList: responsePlansList.discounts,
+          success: true,
+        });
+        setSelectedPlan(0);
+        applyDiscount(safeDiscountId);
```

Comment:
In my opinion these three operations should be done as a single reducer operation.